### ISSUE TITLE
ntopng: interfaces field should be a multi-choice selector

### ISSUE THREAD

**Issue by mihakralj** [opnsense/plugins#2229]
**Describe the bug**
ntopng configuration screen should allow selection of multiple interfaces that are monitored by ntopng service.

**To Reproduce**
Services->ntopng
Select Interfaces selector

**Expected behavior**
Selection of multiple interfaces should be allowed, not just one or all (none == all)

**Screenshots**

![image](https://user-images.githubusercontent.com/31756078/107448996-9f7a7080-6af7-11eb-8517-480ca7c6cc31.png)

**Comment by D3R-ST3FAN** [comment]
I already filed a similar request ( #2096 ).
Any ETA?

Cheers, Stefan

**Comment by mihakralj** [comment]
Let me cook a fix this weekend.
In a nutshell, we need to:
- change the model of ntopng MVC to allow multi-select of interfaces in the form
- change /usr/local/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf to allow multiple lines of **-i=** in the config:

(need to research how to create a loop over all selected interfaces and list them all - currently it executes this:)

{%   if helpers.exists('OPNsense.ntopng.general.interface') and OPNsense.ntopng.general.interface != '' %}
-i={{ physical_interface(OPNsense.ntopng.general.interface) }}

**Comment by mimugmail** [comment]
Have a look at templating of telegraf plugin, there are some loops in there.

**Comment by mihakralj** [comment]
(mental note how tor is doing multiple assignments in torrc using for .. in:)

{% if helpers.exists('OPNsense.tor.general.socks_listen_ip') and OPNsense.tor.general.socks_listen_ip != '' %}
**{%   for listen_interface in OPNsense.tor.general.socks_listen_ip.split(',') %}**
{%     if helpers.exists('interfaces.'+listen_interface+'.ipaddr') %}
{%       set interface_ip = helpers.getNodeByTag('interfaces.'+listen_interface+'.ipaddr') %}
{%       if '.' in interface_ip %}

**Comment by mihakralj** [comment]
OK, I have a patch to fix this:
1. change an interface section in **/usr/local/opnsense/mvc/app/models/OPNsense/Ntopng/General.xml**:
```
        <interface type="InterfaceField">
            <Required>N</Required>
            <Multiple>Y</Multiple>
        </interface>
```
2. change an interface section in **/usr/local/opnsense/mvc/app/controllers/OPNsense/Ntopng/forms/general.xml**:
```
    <field>
        <id>general.interface</id>
        <label>Listening interfaces</label>
        <type>select_multiple</type>
        <help>Select interfaces for ntopng to listen to. Set to none if you want to use all available interfaces.</help>
    </field>
```
3. change an interface section in **/usr/local/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf**:
```
{%   if helpers.exists('OPNsense.ntopng.general.interface') and OPNsense.ntopng.general.interface != '' %}
{%       for listen_interface in OPNsense.ntopng.general.interface.split(',') %}
-i={{ physical_interface(listen_interface) }}
{%       endfor %}
{%   endif %}
```
How to verify that the fix works:
1. the form at Services/Ntopng page allows  selection and save of multiple interfaces
2. multiple interfaces are then stored in /conf/config.xml under <ntopng> branch
3.  multiple line items with "-i " appear in /usr/local/etc/ntopng.conf

**Comment by mihakralj** [comment]
I also added all missing configuration parameters and modified some of the existing ones; this version works with ntopng 4.3 (instructions to get it: https://packages.ntop.org/FreeBSD/)

In the attachment below:

- **models-General.xml** is /usr/local/opnsense/mvc/app/models/OPNsense/Ntopng/General.xml:
- **controllers-forms-general.xml** is /usr/local/opnsense/mvc/app/controllers/OPNsense/Ntopng/forms/general.xml
- **ntopng.conf** is /usr/local/opnsense/service/templates/OPNsense/Ntopng/ntopng.conf

[ntopng.zip](https://github.com/opnsense/plugins/files/6176635/ntopng.zip)

**Comment by mihakralj** [comment]
![image](https://user-images.githubusercontent.com/31756078/111890374-023c1300-89a6-11eb-95c4-08735f5c2a06.png)

**Comment by mihakralj** [comment]
Michael: my suggestion would be to roll a simple fix into **os-ntopng** and to commit enhancements above to **os-ntopng-devel**. Is it possible to change the plugin package dependencies so pkg installs ntopng 4.3 when os-ntopng-devel is selected and installs the old ntopng 3.x when os-ntopng is selected? How?

**Comment by mihakralj** [comment]
I realized that ntop published their own package with ntopng 4.3 - and that plugin has its own GUI wrapper that allows selection of interfaces and (almost) all other attributes that I am showing above. What is missing is forced community and no-logon options.
